metrics-api/src/main/java/software/amazon/swage/metrics/record/DefaultMetricContext.java

```diff
@@ -0,0 +1,46 @@
+package software.amazon.swage.metrics.record;
+
+import software.amazon.swage.collection.ImmutableTypedMap;
+import software.amazon.swage.collection.TypedMap;
+import software.amazon.swage.metrics.Metric;
+import software.amazon.swage.metrics.MetricContext;
+import software.amazon.swage.metrics.Unit;
+
+import java.time.Instant;
+
+final class DefaultMetricContext implements MetricContext {
```

Comment:
Javadoc?

Why does this class exist, etc. etc.  For example: why is this called 'Default' when it's also final?  When would one ever use a different implementation?